### PR TITLE
Fixed crash when searching from omnibox in tor window with search engine extension

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/15140

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/sources.gni [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline chromium_src/chrome/browser/sources.gni:1]: so in general I'm good with this, but the goal here should be to keep deps along with the sources which means we should also have chromium_src/chrome/browser/prefs/sources.gni and then include those sources in the main browser sources here

Comment (bridiver) [reply chromium_src/chrome/browser/sources.gni:1]: or maybe we shouldn't have the extra layer here and just include chromium_src/chrome/browser/prefs/sources.gni  here directly https://github.com/brave/brave-core/pull/8460/files#diff-d00e7b939f2d684a6f82e428ab699596962b739950acc1da8804abb5f607d1b3R13

Comment (bridiver) [reply chromium_src/chrome/browser/sources.gni:1]: also this should include all deps for browser_prefs.cc, not just the new one you added

**Review by bridiver — CHANGES_REQUESTED** [review]
see comments

File: chromium_src/chrome/browser/sources.gni (hunk 1 above)
Comment (simonhong) [reply chromium_src/chrome/browser/sources.gni]: Fixed.

File: chromium_src/chrome/browser/prefs/sources.gni [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import("//components/gcm_driver/config.gni")
```
Comment (iefremov) [inline chromium_src/chrome/browser/prefs/sources.gni:6]: cc @mariospr

**Review by iefremov — APPROVED** [review]

File: chromium_src/chrome/browser/prefs/sources.gni [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+# Fix: Added import loop. Need this import for using |enable_widevine| vars.
```
Comment (bridiver) [inline chromium_src/chrome/browser/prefs/sources.gni:8]: as mentioned in DM we should import brave/browser/sources.gni directly in the existing patch for chrome/browser/BUILD.gn which should help with other circular include issues cc @mariospr

Comment (simonhong) [reply chromium_src/chrome/browser/prefs/sources.gni]: Fixed by https://github.com/brave/brave-core/pull/8460/commits/461f70c943aa984819aa3cf394a7f7dedd5b392a

**Review by bridiver — APPROVED** [review]

**Comment by kjozwiak** [comment]
@simonhong seems like this is still still crashing using `1.25.8 CR: 90.0.4430.61` which should have the above fix. Example of the crashes:

* Uploaded crash report ID: `86180000-9cfd-2306-0000-000000000000`
* Uploaded crash report ID: `85180000-9cfd-2306-0000-000000000000`
* Uploaded crash report ID: `87180000-9cfd-2306-0000-000000000000`

STR:

* install DDG privacy essentials via https://chrome.google.com/webstore/detail/duckduckgo-privacy-essent/bkdgflcldnnnapblkhphbgpggdiikppg?hl=en
* on `brave://extensions/?id=bkdgflcldnnnapblkhphbgpggdiikppg` enable "Allow in private"
* open TOR window
* wait till it's finished initializing/downloading the Tor component
* enter `brave browser` into the omnibox and click `Enter`
* instant crash

I can reproduce 100% of the time on my `Win 10 x64` machine.

**Comment by kjozwiak** [comment]
Created https://github.com/brave/brave-browser/issues/15224 as the above is a different crash and requires actually searching. The above fixes the crash when a user sets focus in the URL.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.25.10 Chromium: 90.0.4430.61 (Official Build) nightly (64-bit)
-- | --
Revision | dced74d4124b26b14126b611853d33512b60c7b6-refs/branch-heads/4430@{#1115}
OS | Windows 10 OS Version 2009 (Build 19042.906)
```

* verified that Brave doesn't crash when running through the cases outlined via https://github.com/brave/brave-browser/issues/15140#issue-851255883
* ensured that Brave doesn't crash when typing into the omnibox within a `Tor window`
* ensured that Brave doesn't crash when typing into the omnibox within a `Private window`

However, https://github.com/brave/brave-browser/issues/15224 is still happening and will need to be fixed in a new PR.

**Pull request merged**